### PR TITLE
Add python-box stubs

### PR DESCRIPTION
**Pull request by JacobHayes**
[Box](https://github.com/cdgriffith/Box) is an arbitrarily nested dict that can also be accessed with attribute (".") lookup. A few things make this trickier to type:
- dynamic attribute lookup/generation with default_box
- name mangled methods (to prevent shadowing assigned values)
- nesting/[type recursion](https://github.com/python/mypy/issues/731)

The upstream package has some inline typing, but largely for "internal" types (ie: method args), not so much the external uses. This is a pretty minimal set of stubs right now, but wanted to see if this seems like the right approach to take. If there's a way around the name manging issue, I might be able to extend type hints upstream/inline if @cdgriffith is open.

### CODE REVIEW

File: stubs/python-box/box.pyi [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from collections.abc import Iterator, MutableMapping
+from typing import Any, Union
+
+class BoxError(Exception): ...
+
+class Box(MutableMapping[str, Union[Any, MutableMapping[str, Any]]]):
```
Comment (JelleZijlstra) [inline stubs/python-box/box.pyi:6]:
```suggestion
class Box(dict[str, Union[Any, MutableMapping[str, Any]]]):
```
It inherits from Dict at runtime, no need to do something different here

**Review by JelleZijlstra — CHANGES_REQUESTED** [review]
This is missing a number of exported names in the package (https://github.com/cdgriffith/Box/blob/master/box/__init__.py), so it should be marked with a module-level `def __getattr__()` to mark it as incomplete.

There's also some missing dunder methods in the Box class, such as `__add__`. You'll need to add those to the class, because `__getattr__` doesn't cover them (checked here: https://mypy-play.net/?mypy=latest&python=3.9&gist=b96ef5e7327755f4d2de2407d8a494af).

**Comment by JelleZijlstra** [comment]
@JacobHayes are you still interested in moving this forward?

**Comment by srittau** [comment]
Closing this for now. It can be reopened if you find time to work on this.

**Pull request closed without merging**